tests/test_update.py

```
@@ -0,0 +1,508 @@
+    ent._handle_coordinator_update()
+    assert ent.available is True
+    assert ent.installed_version == "1_0_0"
+    assert ent.latest_version == "1.0.2"
+    # extra state attributes are created from firmware_update_info keys (top-level ones)
+    assert "opnsense_download_size" in ent.extra_state_attributes
+    assert ent.extra_state_attributes.get("opnsense_download_size") == 123
+    assert ent.extra_state_attributes.get("opnsense_last_check") == 1
+
+
+@pytest.mark.asyncio
+async def test_handle_coordinator_update_upgrade_sets_release_url(
```

Ruff (0.12.2)

Missing docstring in public function

(D103)